aggregation.go

```diff
@@ -0,0 +1,49 @@
+package versionbundle
+
+import (
+	"github.com/giantswarm/microerror"
+)
+
+// Aggregation describes the aggregated version bundles which are wrapped by
+// Capabilities. Each Capability must be within a merged Aggregation must only
+// contain one version bundle. The set of capabilities being bundled within the
+// same list represent the version bundles being able to work together based on
+// the dependencies version bundles defined for their exposed components.
+type Aggregation struct {
```

Comment:
Is that true that always `len(Aggregation.Capabilities[i][j].Bundles) == 1`?
I somehow don't like that. We have additional array that always has a single element. Maybe using a separate type here make sense?

Reply:
That is true. I do not like this either but I did not like the alternatives more. My idea was to have a dynamic and recursive design of the version bundles with as few as possible structures. The current result is that an authority exposes the same structure that is used within the aggregation. This means we can have an arbitrary architecture and combination of authorities within our infrastructure that participate on the versioning scheme. Because of the dynamic and recursive design we can collect, aggregate and forward and collect and aggregate again as many times as we like without needing to care about the aggregation. We can just do it without to worry to create the wrong data structure in the wrong place. Having the constraint you described is not perfect but I can life with that, especially since there is a lot of testing and tooling to work with the version bundles. We can always make sure the aggregation is valid.

Reply:
> Because of the dynamic and recursive design we can collect, aggregate and forward and collect and aggregate again as many times as we like without needing to care about the aggregation.

Do you mean aggregation of aggregations? Is it useful?

I think `Aggregation` could be called a `Distribution`. IMO this is a different thing than capabilities.

> [...] a distribution, or distro, which is collection of software components built, assembled and configured so that it can be used essentially "as is" for its intended purpose.

So I think capabilities allow us to build distributions dynamically. And we shouldn't consider them the same thing. We don't need things like dependencies in there, because there aren't any. We also may want to give them names, maybe other properties, so maybe using the same struct isn't that important.

WDYT?